### PR TITLE
fix(windows): Warn if we reach maximum transient languages

### LINKED ISSUE #3749: bug(windows): Config allows you to install more than 4 transient profiles, but they silently fail to install
**Describe the bug**
Amharic, Sgaw Karen, Hieroglypic, Tibetan and Euro-Latin do not appear in the keyboard list when the Keyman icon on the taskbar is triggered nor the Hotkeys keyboard list. They are all shown in the keyboard list in the Configuration though.

image

**To Reproduce**
1. Install several keyboards and then install the following:
    - https://keyman.com/amharic/
    - https://keyman.com/keyboards/sil_sgaw_karen
    - https://keyman.com/keyboards/hieroglyphic
    - https://keyman.com/tibetan/
    - https://keyman.com/eurolatin/

2. Click on 'Keyman Icon' on the taskbar
3. See error

**Expected behavior**
All installed keyboards should be shown in the keyboard list when the Keyman icon is clicked on.

---

(Delete the irrelevant sections)

**Keyman Desktop:**
 - OS: Windows 10 Pro 2004
 - Keyman version: 14.0.168-alpha
 - Target Application: [self]

**Additional context**
Diagnostic report sent. The case # is 4000.

### LINKED ISSUE #3759: enhancement(windows): Warn user when attempting to install more than 4 transient profiles
On Windows, we are limited to 4 transient profiles (codes 2000, 2400, 2800 and 2C00). Currently, Keyman silently appears to succeed when attempting to install a fifth. This is misleading!

---

@MakaraSok's original report:

For some unknown reason on `Keyman Desktop 14.0.168-alpha`, the keyboard list does not scroll any longer--as a result some keyboards are in the list. The ones missing their corresponding in the keyboard list are highlighted. This happens after a restart.

image

_Originally posted by @MakaraSok in https://github.com/keymanapp/keyman/issues/1843#issuecomment-716341461_

### PR DESCRIPTION
**Pull request by mcdurdin**
Fixes #3749.
Fixes #3759.

Adds a warning dialog when user attempts to add a transient language to a keyboard but the maximum number of transient languages is already installed.

image

If this issue arises when a user is installing a keyboard for the first time, Keyman instead installs the keyboard for the user's default language. This may be a little confusing, but the error condition is difficult to explain and non-technical users will probably be stuck and need to install under their default language in any case. Happy to receive any pushback on this decision.

### CODE REVIEW

**Comment by darcywong00** [comment]
Can you post a screenshot of the warning dialog?

File: windows/src/desktop/kmshell/xml/strings.xml [hunk 1]
```diff
@@ -1068,6 +1068,12 @@ keyboard that you use in Windows.  Keyman keyboards will adapt automatically to
   <!-- String Type: FormatString -->
   <string name="SKKeyboardPartOfPackage" comment="Notice of need to uninstall an entire package">The keyboard '%0:s' is part of package '%1:s'.  You must uninstall the entire package.  Continue?</string>
 
+  <!-- PArameters: %0:s = BCP 47 code -->
+  <!-- Context: Formatted Messages -->
+  <!-- Introduced: 14.0.211 -->
+  <!-- String Type: FormatString -->
+  <string name="SKInstallLanguageTransientLimit" comment="Message shown when Keyman is unable to install a Windows language for a keyboard">Unable to install keyboard language; Windows has a limit of 4 custom 'transient' languages, and you may have reached this limit.</string>
```
Comment (darcywong00) [inline windows/src/desktop/kmshell/xml/strings.xml:1075]: Thoughts about listing the 4 custom 'transient' languages and suggest the user uninstall one of them?

Comment (mcdurdin) [reply windows/src/desktop/kmshell/xml/strings.xml:1075]: Good idea; I'll check how feasible that is.

Comment (mcdurdin) [reply windows/src/desktop/kmshell/xml/strings.xml:1075]: It's certainly possible but it's a bit of plumbing to make it happen; not sure I want to invest the time at this point...

**Review by darcywong00 — APPROVED** [review]
lgtm

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 15.0.9-alpha](https://keyman.com/downloads/releases/alpha/15.0.9)

**Pull request merged**